### ISSUE TITLE
Fix peewee issues with pyright when updating a CharField and then saving it

### ISSUE THREAD

**Issue by benoit74** [openzim/gutenberg#200]
When updating an ORM object (e.g. `book.html_etag = etag`), #191 had to add `# type: ignore` hint because pyright was complaining that a `str` cannot be used to set a `CharField`

This is clearly a peewee issue, maybe linked to the fact that we are using an old version.

To be fixed / reported upstream if not already fixed.

**Comment by Lorrainnn** [comment]
I came across the same issue and checked Peewee’s GitHub — it looks like there’s no existing issue about this.
Do you think it’s still worth reporting upstream, or would it make sense for me to try submitting a fix?
Also, would using `cast(str | None, CharField(...))` be a better workaround than # type: ignore?

This is question somehow very similar to issue #201 .

**Comment by benoit74** [comment]
Asking for help upstream is indeed a good idea, please do, thanks.

**Comment by Lorrainnn** [comment]
Reported to upstream: https://github.com/coleifer/peewee/issues/2980

**Comment by benoit74** [comment]
Maybe worth to give a try to https://pypi.org/project/types-peewee/

**Comment by Lorrainnn** [comment]
> Maybe worth to give a try to https://pypi.org/project/types-peewee/

lol This one works. I am testing on it rn.

**Comment by Lorrainnn** [comment]
I'm not sure why, but this solution seems a bit unstable. I'm currently looking for more type stubs for Peewee, so any suggestions are welcome! :)

**Comment by benoit74** [comment]
>  this solution seems a bit unstable.

what kind of instability?

**Comment by Lorrainnn** [comment]
Yeah, I just meant that Pyright was picking up the types-peewee stubs before,  (literally after 1 night, but now it’s not lmao. Not sure why rn

**Comment by benoit74** [comment]
I will have a look later, probably not in the coming days however, sorry
